PHOENIX-5185 support Math PI function

yanxinyi opened the pull request.

karanmehta93 commented on phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    @Test
+    public void testMathPIFunction() throws Exception {
```
nit: please break this up in 3 tests

karanmehta93 replied:
Sorry, I wasn't much clear earlier. Please put this in three separate Junit `@Test` tests.

karanmehta93 commented on phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java:
```diff
@@ -0,0 +1,70 @@
+public class MathPIFunctionEnd2EndIT extends ParallelStatsDisabledIT {
+    private static final double ZERO = 1e-9;
+
+    private static boolean twoDoubleEquals(double a, double b) {
```
nit: You might wanna move it in a utility class and use it for other tests too.

karanmehta93 commented:
Looks pretty good overall, Nice work @yanxinyi !

yanxinyi commented:
1. split into 3 test cases.
2. moved utility function to BaseTest
3. removed dup utility function on all math function test files and called from utility function inside BaseTest

karanmehta93 commented:
Lets get your PI function patch committed on the PI day. :-)

karanmehta93 commented:
Thanks @yanxinyi for the patch. Can you also put up PR for 4.x-HBase-1.4, 4.x-HBase-1.3 and 4.x-HBase-1.2 branches?
